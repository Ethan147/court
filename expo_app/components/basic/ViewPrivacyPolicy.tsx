import React from "react";
import {
  View,
  Text,
  TextStyle,
  ViewStyle,
  StyleProp,
  Pressable,
} from "react-native";
import { useNavigation, NavigationProp } from "@react-navigation/native";

interface ViewPrivacyPolicyProps {
  onPress?: () => void;
  passStyles?: {
    viewPrivacyPolicyView?: StyleProp<ViewStyle>;
    viewPrivacyPolicyText?: StyleProp<TextStyle>;
  };
}

type ViewPrivatePolicyNavigationProp = NavigationProp<any, "PrivacyPolicy">;

const ViewPrivacyPolicy: React.FC<ViewPrivacyPolicyProps> = ({
  passStyles,
}) => {
  const styles = {
    viewPrivacyPolicyView: passStyles?.viewPrivacyPolicyView || {},
    viewPrivacyPolicyText: passStyles?.viewPrivacyPolicyText || {},
  };

  const navigation = useNavigation<ViewPrivatePolicyNavigationProp>();

  const handlePress = () => {
    navigation.navigate("PrivacyPolicy");
  };

  return (
    <View>
      <View style={styles?.viewPrivacyPolicyView}>
        <Pressable onPress={handlePress}>
          <Text style={styles?.viewPrivacyPolicyText}>View Privacy Policy</Text>
        </Pressable>
      </View>
    </View>
  );
};

export default ViewPrivacyPolicy;
