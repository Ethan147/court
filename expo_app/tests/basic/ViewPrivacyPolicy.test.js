import { useNavigation } from "@react-navigation/native";
import { render, fireEvent } from "@testing-library/react-native";
import React from "react";

import ViewPrivacyPolicy from "../../components/basic/ViewPrivacyPolicy";

// Mock out the useNavigation hook
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: jest.fn(),
}));

describe("ViewPrivacyPolicy", () => {
  it("renders correctly", () => {
    const mockNavigation = { navigate: jest.fn() };
    useNavigation.mockReturnValue(mockNavigation);

    const { getByText } = render(<ViewPrivacyPolicy />);

    expect(getByText("View Privacy Policy")).toBeTruthy();
  });

  it("calls navigation on press", () => {
    const mockNavigation = { navigate: jest.fn() };
    useNavigation.mockReturnValue(mockNavigation);

    const { getByText } = render(<ViewPrivacyPolicy />);
    const policyLink = getByText("View Privacy Policy");

    fireEvent.press(policyLink);

    expect(mockNavigation.navigate).toHaveBeenCalledWith("PrivacyPolicy");
  });

  it("snapshot", () => {
    const tree = render(<ViewPrivacyPolicy />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
