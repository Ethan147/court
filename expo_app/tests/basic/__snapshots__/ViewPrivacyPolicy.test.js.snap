// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPrivacyPolicy snapshot 1`] = `
<View>
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={{}}
      >
        View Privacy Policy
      </Text>
    </View>
  </View>
</View>
`;
